### PR TITLE
Kneth/fix partial sync

### PR DESCRIPTION
**Pull request by blagoev**

### CODE REVIEW

File: tests/js/session-tests.js [hunk 1]
```diff
@@ -803,10 +803,12 @@ module.exports = {
                         schema: [{ name: 'Dog', properties: { name: 'string' } }]
                     };
 
+                    Realm.deleteFile(config);
```
Comment (blagoev) [inline tests/js/session-tests.js]: why do we need to do this. I don't think we have the file there but it does not hurt to leave it just to be sure

File: src/js_realm.hpp [hunk 2]
```diff
@@ -1045,10 +1045,17 @@ void RealmClass<T>::subscribe_to_objects(ContextType ctx, ObjectType this_object
     args.validate_count(3);
 
     SharedRealm realm = *get_internal<T, RealmClass<T>>(this_object);
-    std::string class_name = Value::validated_to_string(ctx, args[0]);
+    std::string object_type = Value::validated_to_string(ctx, args[0]);
```
Comment (blagoev) [inline src/js_realm.hpp]: isn't this a class name

File: src/js_realm.hpp [hunk 3]
```diff
@@ -1045,33 +1045,43 @@ void RealmClass<T>::subscribe_to_objects(ContextType ctx, ObjectType this_object
...
+
+    if (object_schema == schema.end()) {
+        throw std::runtime_error("Object type '" + object_type + "' not found in schema.");
+    }
+
     Protected<ObjectType> protected_this(ctx, this_object);
     Protected<typename T::GlobalContext> protected_ctx(Context<T>::get_global_context(ctx));
+    Protected<FunctionType> protected_callback(ctx, callback);
     auto cb = [=](realm::Results results, std::exception_ptr err) {
+        HANDLESCOPE
```
Comment (kneth) [inline src/js_realm.hpp]: Damn!!!!

**Comment by kneth** [comment]
`react-tests Debug` fails during to simulator issues. It passes locally.

**Pull request merged**